test/test_internet.py

```diff
@@ -26,46 +25,46 @@ def test_get_redirect_url(self):
 
     def test_get_redirect_url_max_hops_exceeded(self):
         url = "http://www.vision.caltech.edu/visipedia-data/CUB-200-2011/CUB_200_2011.tgz"
-        with self.assertRaises(RecursionError):
+        with pytest.raises(RecursionError):
             utils._get_redirect_url(url, max_hops=0)
 
     def test_download_url(self):
         with get_tmp_dir() as temp_dir:
             url = "http://github.com/pytorch/vision/archive/master.zip"
             try:
                 utils.download_url(url, temp_dir)
-                self.assertFalse(len(os.listdir(temp_dir)) == 0)
+                assert len(os.listdir(temp_dir)) != 0
             except URLError:
-                msg = "could not download test file '{}'".format(url)
-                warnings.warn(msg, RuntimeWarning)
-                raise unittest.SkipTest(msg)
+                pytest.skip(f"could not download test file '{url}'")
 
     def test_download_url_retry_http(self):
         with get_tmp_dir() as temp_dir:
             url = "https://github.com/pytorch/vision/archive/master.zip"
             try:
                 utils.download_url(url, temp_dir)
-                self.assertFalse(len(os.listdir(temp_dir)) == 0)
+                assert len(os.listdir(temp_dir)) != 0
             except URLError:
-                msg = "could not download test file '{}'".format(url)
-                warnings.warn(msg, RuntimeWarning)
-                raise unittest.SkipTest(msg)
+                pytest.skip(f"could not download test file '{url}'")
 
     def test_download_url_dont_exist(self):
         with get_tmp_dir() as temp_dir:
             url = "http://github.com/pytorch/vision/archive/this_doesnt_exist.zip"
-            with self.assertRaises(URLError):
+            with pytest.raises(URLError):
                 utils.download_url(url, temp_dir)
 
-    @unittest.mock.patch("torchvision.datasets.utils.download_file_from_google_drive")
-    def test_download_url_dispatch_download_from_google_drive(self, mock):
+    def test_download_url_dispatch_download_from_google_drive(self, mocker):
```

Comment:
I added the `pytest-mock` dependency because it follows a very close API  to what `unittest.mock` has already, so we won't need to change a lot of code, here and in the other files where we use mocking like `test_datasets_utils.py`.

`pytest` does come with a `monkeypatch` fixture by default but it's quite different and for example it doesn't expose some common utilities such as `assert_called_once[_with]`.



To give you an idea, this is what the function would look like with `monkeypatch`:

```py
    def test_download_url_dispatch_download_from_google_drive(self, monkeypatch):
        url = "https://drive.google.com/file/d/1hbzc_P1FuxMkcabkgn9ZKinBwW683j45/view"

        id_ = "1hbzc_P1FuxMkcabkgn9ZKinBwW683j45"
        filename_ = "filename"
        md5_ = "md5"
        n_calls_ = 0

        with get_tmp_dir() as root_:
            def the_mock(file_id, root, filename=None, md5=None):
                assert file_id == id_
                assert filename == filename_
                assert md5 == md5_
                assert root == root_
                nonlocal n_calls_
                n_calls_ += 1
            monkeypatch.setattr('torchvision.datasets.utils.download_file_from_google_drive', the_mock)

            utils.download_url(url, root_, filename_, md5_)

        assert n_calls_ == 1
```
Of course, we could factorize this out and have our own assert_called_once etc., but IMHO it would be overkill and re-inventing the wheel. I think it's worth adding the dependency, which is only for tests anyway. I checked and it's available on fbcode.

Thoughts @pmeier @fmassa @datumbox @prabhat00155 ?

Reply:
Very much in favor of adding `pytest-mock`. It is just a wrapper around `unittest.mock` so it is 100% compatible. It is even [governed by `pytest`](https://github.com/pytest-dev/pytest-mock/) so we don't need to worry about future support or compatibility.